torch/csrc/jit/codegen/cuda/contiguity.cpp

```diff
@@ -520,9 +528,19 @@ void ContigIDs::handle(Merge* merge) {
       root_ids.erase(root_id);
       // If the last id isn't contiguous that's fine, we can use the stride of
       // the last iter domain to multiply the contig index.
-      if (!root_contiguity_[root_id_i] && !root_ids.empty()) {
-        // Otherwise this merge->out() isn't a contiguously indexable ID
-        return;
+      if (!root_contiguity_[root_id_i]) {
+        // If it's the last root, and we're indexing
+        // (!ignore_consistent_ordering) we can still consider this ID
+        // contiguously indexable since it will still be multiplied by its
+        // stride. If we're computing predicates, then we don't want to do this,
+        // as when we mark something as non-contiguous we don't want it merged
+        // with any other domains.
```

Comment:
I don't follow this. First of all, are we always assuming tensors are fully contiguous when generating predicates? That's what I remember, and if it's still the case, this part of the code should never be reachable for predicates.

Reply:
I just tried to split up the conditional into two stages. The predicate path gets caught below in `!ignore_consistent_ordering_` sorry for the confusion. Predicate will have `ignore_consistent_ordering_ == true` so it won't exit/return non-contiguous based on this check.

Reply:
Comment is confusing, let me take another attempt at it.